server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -2073,6 +2183,30 @@ public static Setting<TimeValue> timeSetting(
         );
     }
 
+    private static class TimeValueParser implements Function<String, TimeValue>, Writeable {
```

Comment:
You need to instantiate this at line 2220 to replace the existing static method.